packages/fuel-indexer/src/ffi.rs

```diff
@@ -70,8 +70,23 @@ fn get_object_id(mem: &MemoryView, ptr: u32, len: u32) -> FFIResult<String> {
 }
 
 /// Log the string at the given pointer to stdout.
-fn log_data(mut env: FunctionEnvMut<IndexEnv>, ptr: u32, len: u32, log_level: u32) {
+fn log_data(
+    mut env: FunctionEnvMut<IndexEnv>,
+    ptr: u32,
+    len: u32,
+    log_level: u32,
+) -> Result<(), WasmIndexerError> {
     let (idx_env, store) = env.data_and_store_mut();
+
+    if idx_env
+        .kill_switch
+        .load(std::sync::atomic::Ordering::SeqCst)
+    {
+        // If the kill switch has been flipped, returning an error will cause an
+        // early termination of WASM execution.
+        return Err(WasmIndexerError::KillSwitch);
+    }
```

Comment:
This is definitely fine, but I can't help but wonder if there's a way to de-dupe this code and perhaps have a wrapper function that checks the kill switch before running the desired operation. 🤔